client/rest-high-level/src/main/java/org/elasticsearch/client/ml/dataframe/OutlierDetection.java

```diff
@@ -78,11 +79,21 @@ public OutlierDetection(@Nullable Integer nNeighbors, @Nullable Method method) {
         this.method = method;
     }
 
-    /**
-     * Constructs the default outlier detection configuration
-     */
-    public OutlierDetection() {
-        this(null, null);
+    @Override
+    public String getName() {
+        return NAME.getPreferredName();
+    }
+
+    @Override
+    public Map<String, Object> getParams() {
```

Comment:
This method is only necessary for the backend. We can remove it from `DataFrameAnalysis`.

Reply:
Done.